pkg/config/validator.go

```diff
@@ -72,19 +73,35 @@ type (
 	// AdapterToAspectMapper returns the set of aspect kinds implemented by
 	// the given builder.
 	AdapterToAspectMapper func(builder string) KindSet
+
+	// BuilderInfoFinder is used to find specific handlers BuilderInfo for configuration.
+	BuilderInfoFinder func(name string) (*adapter.BuilderInfo, bool)
+
+	// SetupHandlerFn is used to configure handler implementation with Types associated with all the templates that
+	// it supports.
+	SetupHandlerFn func(actions []*pb.Action, constructors map[string]*pb.Constructor,
```

Comment:
It was mostly a general reaction to reading this file and seeing multiple funcs that take a massive number of args (=> bad code smell IMO).